### DISCUSSION TITLE
Unsure of how to update

### DISCUSSION THREAD

**Discussion by doggie015** [mastodon/mastodon#20911] [Q&A] [answered]
I followed an incorrect guide to set up a mastodon server in a docker container under Ubuntu 20.04 LTS, and the upshot is that, while the server is up, a user known as 991 is running the server but has no home directory and thus no way to update. How do I go about fixing this situation?

**Comment by MrGeneration** [comment, marked as answer]
Your docker host technically does not have to know the said user.
If you'd prefer it does, you can always add the user with the user id 991 - don't forget to do the same for the group if not done automatically.

**Reply by doggie015** [reply]
Cool. So how do I update mastodon if I can’t login as the user running mastodon?

**Reply by MrGeneration** [reply]
Ummm... you are aware how docker works?
You pull the update on the tag version you want (if you're not using latest by default) and follow the actual update instructions of the fitting version you're going to. From 3.5.3 to 4.0.x you'd find that here: https://github.com/mastodon/mastodon/releases/tag/v4.0.0

You technically can skip the pre database migration step if you're willing more downtime (on my instance it wasn't much time effort).
All database migrations you're running are to be run **within** the container.

`docker exec -ti <container id / name> <command>`
You wanna run migrations against the web container.

**Reply by doggie015** [reply]
Alright. Thanks
